_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+---
+layout: blog_post
+title: "How to Deploy a .NET Container with AWS ECS Fargate"
+author: [deepu-sasidharan, chase-aucoin]
+by: advocate
+communities: [.net]
+description: "This is a tutorial on how to secure your .NET containerized chat app with AWS ECS Fargate and Okta."
+tags: [containers, aws, fargate, c-sharp, csharp, dotnet, dot-net, asp-dot-net, vuejs]
+tweets:
+  - "Ever played with AWS Fargate? Learn how to authenticate users to your #dotnetcore #container hosted in #aws ->."
+  - "Unsure about #containerized #dotnet with #Fargate in #AWS? We've got a primer just waiting for you ->"
+  - "Learn how to create secure .NET #containers in #AWS with Fargate ->"
+image: blog/featured/okta-dotnet-mouse-down.jpg
+type: conversion
+---
+
+In a previous article, we learned [how to host a serverless .NET application using AWS Lambda](/blog/2020/06/08/serverless-lambda-functions-csharp). We talked about the history of serverless and how companies are using these types of technology to simplify delivering APIs and functionality faster than traditional methods. Some problems will arise in this type of application when you need more capability than your standard HTTP requests like GET, POST, PUT, DELETE, etc. Web Sockets is a great example of this.
+
+{% include toc.md %}
+
+## Understanding Containers vs. Virtual Machines
+
+Around 2013, [Docker](https://www.docker.com/) was released, beginning a shift in how we think about hosting applications and managing infrastructure. Infrastructure teams and operations have been leveraging Virtual Machines with great success for decades, so you may be tempted to think, "isn't this just a virtual machine?" In spirit, yes; in application, no.
+
+To talk about containers, we have to talk about Virtual Machines. A virtual machine (VM) is a virtual representation of a computer all the way from its boot process to loading an entire operating system. This provides a lot of flexibility because you gain fine-grained control over an entire virtual computer, but it brings all the pitfalls of a whole computer as well. The operating system has to be maintained, patched, and updated; disks have to be managed, and all manner of operational overhead goes into their care and feeding. Notably, VMs are also huge because an entire OS has to go on them. So when you start a VM, you have to go through the entire boot sequence of a traditional computer.
```

Comment:
```suggestion
To talk about containers, we have to talk about virtual machines. A virtual machine (VM) is a virtual representation of a computer, all the way from its boot process to loading an entire operating system (OS). This provides a lot of flexibility. You gain fine-grained control over an entire virtual computer, as well as all the pitfalls of a whole computer. The operating system has to be maintained, patched, and updated; disks have to be managed, and all manner of operational overhead goes into their care and feeding. Notably, VMs are huge because an entire OS has to go on them. So when you start a VM, you have to go through the entire boot sequence of a traditional computer.
```
conciseness, varied sentence length for readability + avoid missing context by removing "it"